### PR TITLE
DOCS-#4241: Update warnings and docs regarding defaulting to pandas

### LINKED ISSUE #4241: Refresh default to pandas info
**Is your feature request related to a problem? Please describe.**
Update warnings and docs regarding defaulting to pandas

### PR DESCRIPTION
**Pull request by Garra1980**
Signed-off-by: izamyati <igor.zamyatin@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4241 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by Garra1980** [comment]
cc @devin-petersohn

File: modin/error_message.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ def default_to_pandas(cls, message=""):
         if not cls.printed_request_implementation:
             message = (
                 "{}\n".format(message)
-                + "To request implementation, send an email to "
-                + "feature_requests@modin.org."
+                + "Please refer to "
```
Comment (YarShev) [inline modin/error_message.py:51]: Given this message, `printed_request_implementation` should be corrected too.

Comment (Garra1980) [reply modin/error_message.py:51]: > Given this message, `printed_request_implementation` should be corrected too.

Fixed, I suppose

File: docs/release_notes/release_notes-0.14.0.rst [hunk 2]
```diff
@@ -12,7 +12,10 @@ Key Features and Updates
   * FIX-#4162: Use `skipif` instead of `skip` for compatibility with pytest 7.0 (#4163)
   * FIX-#4158: Do not print OmniSci logs to stdout by default (#4159)
   * FIX-#4177: Support read_feather from pathlike objects (#4177)
+  * FIX-#3442: Add support for dt.hour in omnisci (#4215)
```
[the remark this answers is not in the document]
Comment (Garra1980) [reply docs/release_notes/release_notes-0.14.0.rst:15]: Bah... :))))

File: modin/error_message.py (hunk 1 above)
Comment (Garra1980) [reply modin/error_message.py:51]: fixed

File: docs/release_notes/release_notes-0.14.0.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Garra1980) [reply docs/release_notes/release_notes-0.14.0.rst]: Removed

File: docs/release_notes/release_notes-0.14.0.rst [hunk 3]
```diff
@@ -13,6 +13,7 @@ Key Features and Updates
   * FIX-#4158: Do not print OmniSci logs to stdout by default (#4159)
   * FIX-#4177: Support read_feather from pathlike objects (#4177)
   * FIX-#4234: Upgrade pandas to 1.4.1 (#4235)
+  * FIX-#4241: Update warnings and docs regarding defaulting to pandas (#4242)
```
[the remark this answers is not in the document]
Comment (Garra1980) [reply docs/release_notes/release_notes-0.14.0.rst]: Done

**Comment by Garra1980** [comment]
@devin-petersohn CI is green, any more comments?

**Review by YarShev — APPROVED** [review]

File: docs/release_notes/release_notes-0.14.0.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (YarShev) [reply docs/release_notes/release_notes-0.14.0.rst]: Do not forget to add this as part of a separate PR 😄

**Comment by YarShev** [comment]
> @devin-petersohn CI is green, any more comments?

I believe everything is okay.

**Comment by YarShev** [comment]
If there is something that we need to add/correct, feel free to ask/reopen a PR.

**Pull request merged**